cocos/2d/CCFontFNT.cpp

```diff
@@ -531,13 +572,26 @@ FontFNT * FontFNT::create(const std::string& fntFilePath, const Vec2& imageOffse
     return tempFont;
 }
 
-FontFNT::FontFNT(BMFontConfiguration *theContfig, const Vec2& imageOffset /* = Vec2::ZERO */)
-:_configuration(theContfig)
-,_imageOffset(CC_POINT_PIXELS_TO_POINTS(imageOffset))
+FontFNT* FontFNT::create(const std::string& fntFilePath, const Vec2& imageOffset)
+{
+    return create(fntFilePath, Rect(imageOffset.x, imageOffset.y, 0, 0), false);
```

Comment:
Why it is `Rect(imageOffset.x, imageOffset.y, 0, 0)` ? The size is 0?

Reply:
The current API uses a Vec2, which is the X and Y offset of the bitmap font image in the sprite sheet.  In order to support rotation in a sprite sheet (90° to the right for PLIST sprite sheets), it needs x, y, width, height.  Now in adapting the old API to the new API, the passed in Vec2 imageOffset, we can just use a Rect with the image offset X,Y and width/height both 0, since it works exactly the same way.  

Any code using the Vec2 imageOffset implies that the sprite sheet does not have rotated textures.  There is a test in cpp-tests that shows this, under the "Old Label API".  So, this gets converted to Rect with width/height 0, and the code path does exactly the same thing as before (it doesn't use width/height if the texture isn't rotated).